scripts/ci-build-azure-ccm.sh

```diff
@@ -87,6 +111,12 @@ can_reuse_artifacts() {
         echo "false" && return
     fi
 
+    for BINARY in azure-acr-credential-provider azure-acr-credential-provider.exe; do
```

Comment:
can we skip this if `TEST_ACR_CREDENTIAL_PROVIDER` is not true? otherwise it will cause every job to rebuild artifacts for CCM and CNM if they previously skipped building and pushing the azure-acr-credential-provider binaries even though they don't need them

Reply:
Makes sense. Updated.